File: load-template/python/src/window.py
from gi.repository import Gtk, Adw

@Gtk.Template(resource_path="/org/gtk/Example/window.ui")
class Window(Adw.ApplicationWindow):
    __gtype_name__ = "Window"
    button = Gtk.Template.Child()

    def __init__(self, **kwargs):
        # Call __init__ on parent:
        super().__init__(**kwargs)
        # Connect to "clicked" signal of `button`
        self.button.connect("clicked", self.on_button_clicked)

    def on_button_clicked(self, button):
        # Set the label to "Hello World!" after the button has been clicked on:
        button.set_label("Hello World!")

File: load-template/gjs/src/window.js
import Adw from 'gi://Adw?version=1';
import GObject from 'gi://GObject';
import Gio from 'gi://Gio';

const Window = GObject.registerClass({
    GTypeName: 'Window', // `GTypeName` needs to match `class` attribute of the template tag
    // Load from file using relative path (I couldn't find a way to use "resource:///org/gtk/Example/window.ui"):
    Template: Gio.File.new_for_path("./src/window.ui").get_uri(),
    Children: [
        'button' // referring to a GtkButton with the `id` "button"
    ],
}, class extends Adw.ApplicationWindow {
    constructor(params = {}) {
        // Call parent constructor
		super(params);
        // Connect to "clicked" signal of `button`
        this.button.connect("clicked", (button) => {
            // Set the label to "Hello World!" after the button has been clicked on
            this.button.set_label("Hello World!");
        });
	}
});

export default Window;